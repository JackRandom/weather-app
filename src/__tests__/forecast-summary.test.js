import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../components/forecastSummary";


describe("ForecastSummary", () => {

  it("renders correctly", () => {
    const { asFragment } = render(
      
      <ForecastSummary
        date={1525046400000}
        temperature="mockTemperature"
        description="mockDescription"
        icon="800"
        handleForecastSelect={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

it("renders the correct props", () => {
    const { getByTestId} = render(
      <ForecastSummary
        date={1525046400000}
        temperature="mockTemperature"
        description="mockDescription"
        icon="800"
        handleForecastSelect={() => {}}
      />
    );
//handleForecastSelect test

    expect(getByTestId("date-id")).toHaveClass("forecast-summary__date");
    expect(getByTestId("temperature-id")).toHaveClass("forecast-summary__temperature");
    expect(getByTestId("description-id")).toHaveClass("forecast-summary__description");
    expect(getByTestId("icon-id")).toHaveClass("forecast-summary__icon");
  });
});