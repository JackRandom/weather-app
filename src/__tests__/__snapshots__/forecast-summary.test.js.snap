// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <h2>
    <div
      class="forecast-summary__date"
      data-testid="date-id"
    >
      <span>
         Mon 30th Apr
      </span>
    </div>
    <div
      class="forecast-summary__temperature"
      data-testid="temperature-id"
    >
      <span>
         mockTemperatureÂ°c 
      </span>
    </div>
    <div
      class="forecast-summary__description"
      data-testid="description-id"
    >
      <span>
         mockDescription 
      </span>
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="icon-id"
    >
      <span>
          
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
         
      </span>
    </div>
    <button>
      More Details
    </button>
  </h2>
</DocumentFragment>
`;
